.github/workflows/CI.yml

```
@@ -36,9 +36,12 @@ jobs:
         uses: actions/cache@v3
         with:
           path: |
-            ~/.cargo/
-            ./target/
-          key: "cache-OS:${{ matrix.os }}-Rust:${{ matrix.rust }}-${{ hashFiles('./Cargo.toml') }}-Cpp:${{ matrix.cpp }}"
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: "cache-OS:${{ matrix.os }}-Rust:${{ matrix.rust }}-${{ hashFiles('Cargo.lock') }}-Cpp:${{ matrix.cpp }}"
```

You have to check in `Cargo.lock` if you want to rely on it.

```suggestion
          key: "cache-OS:${{ matrix.os }}-Rust:${{ matrix.rust }}-${{ hashFiles('Cargo.toml') }}-Cpp:${{ matrix.cpp }}"
```